crates/ntx-builder/src/state/mod.rs

```diff
@@ -82,15 +83,22 @@ impl State {
             nullifier_idx: BTreeMap::default(),
         };
 
-        let notes = state.store.get_unconsumed_network_notes().await?;
-        for note in notes {
-            let prefix = note.account_prefix();
-
-            // Ignore notes which don't target an existing account.
-            let Some(account) = state.fetch_account(prefix).await? else {
-                continue;
-            };
-            account.add_note(note);
+        for prefix in self.accounts.iter() {
+            let network_account_id_prefix = NetworkAccountPrefix::try_from(0u32).unwrap();
```

Comment:
@Mirko-von-Leipzig its this part where I don't know that we could invert the note->account mapping and go account->notes